### PR TITLE
More Jokes API

### PR DESCRIPTION
**Pull request by Yash17Agrawal**
Fixed issue regarding #1

### CODE REVIEW

**Comment by 526avijitgupta** [comment]
@Yash17Agrawal There are merge conflicts that need to be resolved. Run the following commands:

``` bash
git branch # Verify that the current branch is master
git remote add upstream https://github.com/osdc/osdc-bot
git fetch upstream
git merge upstream/develop
# Resolve merge conflicts if there are any
git add .
git commit -m "Update branch and resolve merge conflicts" # Or any other commit message
git push origin master
```

Please make sure to verify that the bot is functioning correctly before committing and pushing your code in the final step!

**Comment by Yash17Agrawal** [comment]
@526avijitgupta  sir shouldnt it be git merge upstream/master ?

File: constants.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ const CHATROOM_URL = `https://api.gitter.im${CHATROOM_SUFFIX_URL}`;
 const SERVER_PREFIX_URL = 'http://127.0.0.1:5000';
 const SERVER_DEPLOY_URL = `${SERVER_PREFIX_URL}/deploy`;
 const SERVER_HOWDOI_PREFIX_URL = `${SERVER_PREFIX_URL}/howdoi?query=`;
-const JOKES_API_URL = 'http://api.icndb.com/jokes/random';
+const JOKES_API_URL =[ 'http://api.icndb.com/jokes/random','http://tambal.azurewebsites.net/joke/random'];
```
Comment (yashmehrotra) [inline constants.js]: Fix Linting, also, convert this list into a list of Objects, where we can store url and method to access the joke.

File: services/jokeService.js [hunk 2]
```diff
@@ -4,10 +4,15 @@ const constants = require('../constants');
 
 module.exports = {
   getJoke: (callback, username) => {
-    request(constants.JOKES_API_URL, (error, response, body) => {
-      if (!error && response.statusCode === 200) {
-        return callback(JSON.parse(body).value.joke, username);
-      }
+    var index = Math.floor(Math.random() * constants.JOKES_API_URL.length);
+    var randomQuote =  constants.JOKES_API_URL[index];
+    request(randomQuote, (error, response, body) => {
```
Comment (yashmehrotra) [inline services/jokeService.js]: Change name to `randomJoke`

File: services/jokeService.js [hunk 3]
```diff
@@ -4,10 +4,15 @@ const constants = require('../constants');
...
+      if (!error && response.statusCode == 200) {
+	if (index==0)
```
Comment (yashmehrotra) [inline services/jokeService.js]: Same as the comment on `constants.js`, instead of `if/else` the access method should be there in `const JOKE_API_URL`

**Review by yashmehrotra — CHANGES_REQUESTED** [review]

File: services/jokeService.js (hunk 3 above)
Comment (Yash17Agrawal) [reply services/jokeService.js]: okay @yashmehrotra

**Comment by Yash17Agrawal** [comment]
@526avijitgupta @yashmehrotra  Please review.

File: constants.js (hunk 1 above)
Comment (yashmehrotra) [reply constants.js]: @Yash17Agrawal convert this list into a list of Objects, where we can store url and method to access the joke.

File: services/jokeService.js [hunk 4]
```diff
@@ -4,9 +4,15 @@ const constants = require('../constants');
 
 module.exports = {
   getJoke: (callback, username) => {
-    request(constants.JOKES_API_URL, (error, response, body) => {
+    const index = Math.floor(Math.random() * constants.JOKES_API_URL.length);
+    const randomQuote =  constants.JOKES_API_URL[index];
```
Comment (526avijitgupta) [inline services/jokeService.js]: There seems to be an extra space after `=`

File: services/jokeService.js [hunk 5]
```diff
@@ -4,9 +4,15 @@ const constants = require('../constants');
...
+    request(randomQuote, (error, response, body) => {
       if (!error && response.statusCode === 200) {
-        return callback(JSON.parse(body).value.joke, username);
+        if (index === 0) {
+          return callback(JSON.parse(body).value.joke, username);
+        } else {
+          return callback(JSON.parse(body).joke, username);      
+        }
```
Comment (526avijitgupta) [inline services/jokeService.js]: Better to do:

``` js
const jokeText = (
  (index === 0) ? JSON.parse(body).value.joke : JSON.parse(body).joke);
return callback(jokeText, username);
```

**Review by 526avijitgupta — CHANGES_REQUESTED** [review]
@Yash17Agrawal Minor changes requested. Overall looks good!

**Review by 526avijitgupta — APPROVED** [review]
LGTM! @yashmehrotra -- anything from your side ?

**Comment by yashmehrotra** [comment]
@526avijitgupta Using `if (index == 0) {} else {}` is wrong, its better if we store the access method in the list, something like,
```
[
  {
     'url': 'http://abc.def',
     'joke_val': 'response.body.value.joke'
   },
    {
    }
]
```

**Comment by yashmehrotra** [comment]
@526avijitgupta LGTM
@Yash17Agrawal Good work!

**Pull request merged**